Bonner checkmark profile

MImran2002 opened the pull request:
## Bonner requirements checkmarks on the profile page 

Fixes   #1370

- The checkmarks for Bonner requirements on the profile page should respect the frequency of the requirement. Something that is required once, would have (1/1) or (0/1). Something that is required twice should have (0/2) or (1/2) or (2/2), every term should show the numbers of time they have done the certification against the total expected certification based on their class level. For example if they are Junior they will likely have (0-5/5) or (0-6/6).

## Changes
- Add options for both term and twice in Bonner Management under Brian as admin.
- Reproduce the certification requirement with counts (1/1) if they have done and (0/1) if they have not (This may vary based on frequency.

## Edge-cases to consider before testing:
- These edge case are because the numbers populated are based on two things: the current term and the student's rawClassLevel.
- If any of the cases disrupt these two things it will display no data or wrong data. 
- Cases to avoid:
  1. If you create a new event on top of an existing event that has the same term and the same frequency, only one of those events will appear.
  
  2. Another scenario occurs when the current term is Fall 2023, but an event is created in Spring 2024, and a student is marked as attended for that Spring 2024 event. In this case, the student is being assigned attendance for a future term, so it will not appear in the count.
  
  3. This logic also applies to the past. For example, if a student is a sophomore and the current term is Spring 2023, the list only populates terms from Spring 2023 back to their Freshman year (Spring 2023, Fall 2022, Spring 2022, Fall 2021).
  Any events created before or after these term ranges will not appear in the counts.

  4. Another case is where if the event is created during summer terms it will not count, as certification events do not happen during summer and if the current term is summer and you hover over the view it will show the admin the Fall term  instead of Summer term as the idea is to inform the admin of certification being needed for Fall term as there's no certification for the Summer term. 

# Testing
## Navigating to the branch
- Use git checkout bonnerCheckmarkProfile
- Pull the code with git pull
- Reset the database with to test and run tests/run_tests.sh
- Run the application with flask run
- Make your way to the navigate bar and click on admin drop down feature
- Click Bonner_Management and go to Requirements and click on the "Add Requirement" button then name it and choose every term as the Frequency
- Note:  Make sure that the term that you picked for the event when creating it is a term at which the cohort you picked will be attending school (not graduated at that time ) and not earlier than the current term. Otherwise, it won't show up.

## Creating and checking the certification
***All of the testing must make sure that the term the event is created in because the event must be created between the current term and the term they came to Berea College (you can see the current term in the setting > term management or when you create an event the term that is auto selected is the current term)***
- Go to Create Event and Select Bonner Scholars to create certification events
- Create an Event and make sure you choose the Bonner Requirement that you added a while ago.
- Select the invite cohorts that has Zach Neill.
- And click Create Event
- Go to manage volunteers where the RSVP and Waitlist checked Zach Neill as attended and saved
- Then go to the side bar and navigate to Zach Neill as a user
- Go to My Profile and click the Bonner Program dropdown
- You will see the tick if you have finished and a number of cross on the terms you have not
- Hover over the requirement that you created and will see the tooltips

MImran2002 committed:
term missed and term added

MImran2002 committed:
almost there

MImran2002 committed:
The functions are done I just need to work on the test suite

MImran2002 committed:
remove duplicate certifications, add tooltips to show what terms the student has attended

MImran2002 committed:
final push

bledsoef committed:
Merge branch 'development' into bonnerCheckmarkProfile

WackyWeaver committed:
Merge branch 'development' into bonnerCheckmarkProfile

WackyWeaver reviewed (commented):
There is a html logic issue on the tool tip when there have been no attended terms. Also, the tooltip never shifts when there are fewer terms available for underclassmen.

MImran2002 committed:
change the tooltips location, add an alternative message if no terms are attended and the term will only show if the admin is false

MImran2002 committed:
finding bug

MImran2002 committed:
done

stevensonmichel committed:
Made changes in code structure

stevensonmichel committed:
Merge branch 'development' into bonnerCheckmarkProfile

MImran2002 committed:
user current term is refactored as g.current_term and have hidden the bonner program accordion if the person who we click the user progile is not a student

stevensonmichel committed:
Fixed merge conflicts

MImran2002 committed:
The current term is updated to false and a new current term is created to fit the test case underneath the app.app_context()

MImran2002 committed:
merging

MImran2002 committed:
merge conflicts and probnlem solved

MImran2002 committed:
Merge branch 'development' of https://github.com/BCStudentSoftwareDevTeam/celts into bonnerCheckmarkProfile

WackyWeaver reviewed (dismissed):
LTGM! Please add a comment for the formula in missedTerms and rename certs

stevensonmichel committed:
Fixed logic efficiency with set

stevensonmichel committed:
Fixed issues from test_certification.py

stevensonmichel committed:
removed print statements

WackyWeaver reviewed (dismissed):
I appreciate the comments added to the calculating logic section.  A little long compared to a lot in the rest of the code base, but everything else LGTM!

RueHaile committed:
Merge branch 'development' into bonnerCheckmarkProfile

MImran2002 committed:
Merge branch 'development' of https://github.com/BCStudentSoftwareDevTeam/celts into bonnerCheckmarkProfile

MImran2002 committed:
Merge branch 'bonnerCheckmarkProfile' of https://github.com/BCStudentSoftwareDevTeam/celts into bonnerCheckmarkProfile

MImran2002 committed:
placing change for tick

MImran2002 committed:
userprofile

MImran2002 committed:
stashing

bakobagassas commented:
When I follow the steps in the PR, everything works up to where I click to save the event. It is failing to save the event.

bakobagassas commented:
Also, I have noticed that when I try to create an event, before even selecting the requirement that had "all term", I am unable to select a term in the event itself.

bakobagassas commented:
Also, something seems to be breaking. When I try to access event lists (where we see the event we created), the page is not accessible on your branch; it gives an error page. nb: I already reset the database multiple times.

MImran2002 committed:
remove print statement

MImran2002 commented:
> Also, something seems to be breaking. When I try to access event lists (where we see the event we created), the page is not accessible on your branch; it gives an error page. nb: I already reset the database multiple times.

There are tests breaking in development:
tests/code/test_serviceLearningCourses.py ...F......
tests/code/test_spreadsheet.py F..FFFFFFFFFFF
tests/code/test_term.py ...
tests/code/test_transcripts.py .F..

And for the event I am able to select the term:
<img width="967" height="764" alt="image" src="https://github.com/user-attachments/assets/f8df7ddd-3978-4baf-b598-55b019e98f98" />
 and I am able to see an event list : 
<img width="1071" height="491" alt="image" src="https://github.com/user-attachments/assets/0a273cf6-b0a5-4930-a790-3d43d180ea12" />

I can definitely guide you with the set up when might you be free on Wednesday or Thursday.

bakobagassas commented:
I wonder why it is only showing 3 checkmarks. Wasn't it supposed to be every term (so 8 terms)? 
<img width="1920" height="1032" alt="image" src="https://github.com/user-attachments/assets/447893d0-f7e3-4b72-87a4-38db2552d058" />

MImran2002 commented:
> I wonder why it is only showing 3 checkmarks. Wasn't it supposed to be every term (so 8 terms)? <img alt="image" width="1920" height="1032" src="https://private-user-images.githubusercontent.com/194523576/503794366-447893d0-f7e3-4b72-87a4-38db2552d058.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEwODAxMzcsIm5iZiI6MTc2MTA3OTgzNywicGF0aCI6Ii8xOTQ1MjM1NzYvNTAzNzk0MzY2LTQ0Nzg5M2QwLWY3ZTMtNGI3Mi04N2E0LTM4ZGIyNTUyZDA1OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyMVQyMDUwMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMzAyMzBiZjE0OTAwZmI3MmY2Njk1OGI0NWI5ODc3YzVjNTg3Yzk0NjBlYzgwNjg4NDBiY2QzOTdhNDQwOWRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Batuvhs3YkdgeYYCLA-h8RtzPnba01l9NtcCYE275HE">

Hello Sara, Thank you for the question. That's what I thought as well but after reading through the issues and code I remember why. These are the few questions you might potentially have reviewing this issue.

Why am I seeing only three checkmarks, shouldn't it be eight checkmark?
    The reason you see three checkmarks instead of eight is because the student is a sophomore and it is either a summer term or a spring term. So the student who is a Bonner scholar should only at that point have three checkmarks or ticks because it is populated based on the terms that they are Bonner scholars. Not the overall term this is to address multiple scenario, if a Bonner student decides to drop out of the program there won't be unnecessary populated terms, if they decide to graduate early, the populating of the checkmark will not show a total checkmark at all but only show the checkmarks when they are Bonner scholars and importantly the CELTS department doesn't care about future certification requirment their concern more is if the student have done the previous and present requirement. That is why the populating of checkmark is based on every term they stay in the program or college.

Why is Summer Term move forward to Fall term?
    The reason we have a condition where we manually assigned the current term if its summer term to fall term all the time is because if the celts admin is checking this during the summer term we don't want summer term to be considered towards the overall term which will make it from maximum 8 checkmark to 11 checkmark and no Bonner program student is present during summer so it is more reasonable to show that there is a cross for next term then the summer term to always inform the admin about it. 

I hope these questions might help and if you want to confirm the logic of it you can definitely refer to the test where the logic is tested in multiple scenario with multiple class levels and terms.

bakobagassas reviewed (dismissed):
I had questions about whether the checkmarks represent the number of terms from the past or the total number of terms the person needs to spend in Berea. It seems to be the number of terms the person has already spent in Berea including the present. If this is what the client requires then everything looks good!

BrianRamsay commented on app/logic/certification.py:
```diff
@@ -1,14 +1,68 @@
 from app.models.eventParticipant import EventParticipant
+from app.models.user import User
+
+def termsAttended(certification=None, username=None):
```
We should handle if username is not provided in a different way. Currently, we'll get an error, as `len(attendance)` will break in some fashion, as attendance won't have been defined if there is no username.

We should probably check for a missing username and exit the function right at the top if it is not provided.

Since we are handling that case, we should also have a test case when the expected parameters are not given, and make sure that the function fails gracefully (or fails the way we would expect)

BrianRamsay replied:
Doesn't appear to be resolved. If `username` is `None`, `attendance` will be undefined when we try to use it on line 21.

(username if conditional check has been removed-Imran)

MImran2002 replied:
username conditional check has been removed.

BrianRamsay commented on app/logic/certification.py:
```diff
@@ -1,14 +1,68 @@
+    - If the user's classification is None, assume just 1 expected term.
+    - Subtract the number of terms the student has attended from the expected total to get the missed count.
+    '''
+    classLevel = ["Freshman", "Sophomore", "Junior", "Senior"]
```
We should validate our parameters here, so we have more control over how the function handles missing data. Right now, we'll just get a `NotFoundException` when we try to run the User query, if it is missing.

BrianRamsay commented on app/logic/certification.py:
```diff
@@ -1,14 +1,68 @@
+        currentDescription = currentTerm.description
+
+    for level in range(4):
+        user = User.select().where(User.username == username).get()
```
We don't need to get the user every time in the loop, we can get it before the loop. In this particular case, we can get the user at the top of the function to validate the `username` parameter.

BrianRamsay commented on app/logic/certification.py:
```diff
@@ -1,14 +1,68 @@
+    currentDescription = currentTerm.description
+    
+    # looking into a scenario where the current term is summer so that we can reassigned the current term variable to the next term     
+    if currentTerm.isSummer == True:
```
`isSummer` is already a boolean, so we don't need to compare it to `True`

BrianRamsay commented on app/logic/certification.py:
```diff
@@ -1,14 +1,68 @@
+        currentDescription = f'Fall {currentTerm.year}'
+        currentTerm = Term.select(Term).where(Term.description == currentDescription).get()
+    else:
+        currentDescription = currentTerm.description
```
This case is already handled by the variable initialization on line 40

BrianRamsay commented on app/logic/certification.py:
```diff
@@ -1,14 +1,68 @@
+        elif user.rawClassLevel == classLevel[level]:
+            missedTerms = ((level + 1) * 2) - 1
+        elif str(user.rawClassLevel) == "None":
+            missedTerms = 1
```
Let's use `totalTerms` for this value, because it's not really a missed term until the calculation happens below.

ojmakinde reviewed (changes requested):
I think this approach is a bit unintuitive and unclear.
<img width="400" height="159" alt="image" src="https://github.com/user-attachments/assets/f3f797d4-0115-411b-83b8-2ff69a124583" />
The X's don't fit with the rest of the text and don't tell users very much. Additionally, for events with multiple requirements, this is going to be hard to interact with on mobile, and our users don't want to count the number of events but see it at a glance instead. 

An alternative approach that I recommend would be using an approach like, 
Requirement Name (Frequency Attended / Required Frequency) and using a [tooltip](https://getbootstrap.com/docs/4.0/components/tooltips/) to show the number of events attended. An example of this would be
<img width="225" height="86" alt="image" src="https://github.com/user-attachments/assets/6aa67140-35a8-47bf-a348-ff36de602991" />

Next to the (1/1), add the suggested tooltip and show information for the attended events and the semester, just like we do in the volunteer view. 
<img width="316" height="160" alt="image" src="https://github.com/user-attachments/assets/0f48fddf-f9b4-4164-8e2e-9674a5bf2155" />

MImran2002 committed:
Merge branch 'development' of https://github.com/BCStudentSoftwareDevTeam/celts into visualizeReportData
merging development into visualizeReportData

MImran2002 committed:
still working on bonner checkmark profile

MImran2002 committed:
worked on test suite, new function created jinja template

MImran2002 committed:
new changes and logic'

MImran2002 committed:
new changes after Jame's review

MImran2002 committed:
minor changes

MImran2002 committed:
minor changes

MImran2002 committed:
final changes hopefully

MImran2002 committed:
changes terms toe vents

MImran2002 committed:
added the view and did some jinja templating

MImran2002 committed:
added the view and did some jinja templating

MImran2002 committed:
added the view and did some jinja templating

ojmakinde committed:
Adjusted structure of termsInTotal() function to make it more readable

ojmakinde reviewed (dismissed):
PR looks good. Requested changes have been addressed and it looks much better than it did previously.

BrianRamsay commented on app/logic/certification.py:
```diff
@@ -1,14 +1,102 @@
+    fallAndSpringTerms = []
+    
+    classLevelToYears = {"Freshman": 1, "Sophomore": 2, "Junior": 3, "Senior": 4}
+    yearsCompleted = classLevelToYears.get(studentStanding)
```
This will fail as soon as we are looking at a student who doesn't have their class level reported properly. We need to handle that case in some fashion.

```
mysql> select rawClassLevel, count(*) from user where isStudent group by rawClassLevel;
+---------------+----------+
| rawClassLevel | count(*) |
+---------------+----------+
| NULL          |     1854 |
| Junior        |      383 |
| Freshman      |      503 |
| Sophomore     |      347 |
| Senior        |      276 |
| Graduating    |       73 |
| Non-Degree    |       33 |
+---------------+----------+
```

BrianRamsay replied:
(probably we should just count them as seniors, or whatever we need to do to give them the max number of terms)

MImran2002 replied:
have put an elif condition that checked these rawClassLevel and populated 8 potential terms for the user

MImran2002 committed:
worked on the username and other rawclasslevel possibility

MImran2002 committed:
Merge branch 'bonnerCheckmarkProfile' of https://github.com/BCStudentSoftwareDevTeam/celts into bonnerCheckmarkProfile

MImran2002 committed:
function created and working

ojmakinde commented on app/logic/certification.py:
```diff
@@ -1,14 +1,84 @@
+                    year = startYear + (k // 2) + 1
+                totalTerms.append(f"{season} {year}")
+            break
+        elif user.rawClassLevel == "NULL" or user.rawClassLevel == "Graduating" or user.rawClassLevel == "Non-Degree":
```
Should non-degree folks have any number of semesters? Can you confirm with Brian that they count as students?

MImran2002 replied:
From what I remember Brian showed me that that table he mentioned is underneath student which is why I have to consider that and he told me becasue all NULL, graduating and Bon-degree should populate like seniors @BrianRamsay please correct me if I am wrong. But I will double check with him else well in person.

BrianRamsay replied:
We can treat them as seniors, though there wouldn't be any non-degree Bonner students, so we don't need a specific case for it, really. We just need to not break.

MImran2002 replied:
I am going to resolved this as, the code is considering all the edge cases as seniors.

ojmakinde commented on app/logic/certification.py:
```diff
@@ -1,14 +1,84 @@
+    missedTerms = [term for term in totalTerms if term not in attendedTerms]
+    return missedTerms
+
+def termsInTotal(username):
```
Can you add a docstring here? The function doesn't return all the terms in total. It only returns the Fall + Spring and I think it's important to make that distinction in the comment. Also, I think the previous approach to calculating the terms was cleaner, because this is a bit hard(er) to understand, unless you disagree

MImran2002 replied:
> Yes, the reason I created this termsintotal is because for me this is easier to find the actual start year then populating it from there.  Especially when i was creating test cases I realize it always skipped a term, the count may be right but it never considers the current so I created this function that will find the start year and from there populate the term. I will add a docstring to further clarify it.

MImran2002 committed:
added docstring

ojmakinde commented on app/logic/certification.py:
```diff
@@ -36,6 +35,20 @@ def termsMissed(certification, username):
+    This function calculates all the terms a student have enrolled based on their class level.
+    This is under the impression that as Bonnor Scholars they start their term in Fall and end in four year along with Summer not being counted.
+    Thus, if the current term is summer it is changed to Fall for display purpose.
+    The logic is as follows:
```
I'm sorry, but this is far too long. All I expected was a line or two describing this function and returning the groups, not a description of the logic

MImran2002 replied:
I am sorry, I have reduced it to two lines of explanation. Please tell me if this will work.

ojmakinde replied:
Looks good! Thanks

MImran2002 committed:
added updated docstring

ojmakinde reviewed (changes requested):
I pulled from prod-backup and searched up Syakira's profile. Syakira is a bonner that has a rawClassLevel of null. Here's the SQL query:

```
select user.username, user.rawClassLevel from user inner join bonnercohort on user.username=bonnercohort.user_id where rawClassLevel is NULL and user.username LIKE '%wij%';
+----------+---------------+
| username | rawClassLevel |
+----------+---------------+
| wijayas  | NULL          |
+----------+---------------+
1 row in set (0.001 sec)
```

Here's her bonner profile page:

<img width="1546" height="678" alt="Image" src="https://github.com/user-attachments/assets/bf9be0fc-5f37-44ad-8459-5348193f7ede" />


Also, here's Chase Ackerman's profile:
<img width="1513" height="530" alt="image" src="https://github.com/user-attachments/assets/ce90dbc7-bf78-41f9-8292-d5d800334ac4" />

RueHaile committed:
currently working for all except null status

RueHaile commented:
> I pulled from prod-backup and searched up Syakira's profile. Syakira is a bonner that has a rawClassLevel of null. Here's the SQL query:
> 
> ```
> select user.username, user.rawClassLevel from user inner join bonnercohort on user.username=bonnercohort.user_id where rawClassLevel is NULL and user.username LIKE '%wij%';
> +----------+---------------+
> | username | rawClassLevel |
> +----------+---------------+
> | wijayas  | NULL          |
> +----------+---------------+
> 1 row in set (0.001 sec)
> ```
> 
> Here's her bonner profile page:
> 
> <img alt="Image" width="1546" height="678" src="https://private-user-images.githubusercontent.com/122561188/517069137-bf9be0fc-5f37-44ad-8459-5348193f7ede.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM3Mzc0MjksIm5iZiI6MTc2MzczNzEyOSwicGF0aCI6Ii8xMjI1NjExODgvNTE3MDY5MTM3LWJmOWJlMGZjLTVmMzctNDRhZC04NDU5LTUzNDgxOTNmN2VkZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMVQxNDU4NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNDY3NWFmMDAxNmUyYTgyZTNkYjk5ZTQwNGRkMzg2ZDg4OTcyNjk5M2IyYzhjNTExY2YxZDU2ZmUyODZkMWNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.95LKRKVueFQHNl2oisgXZOGTOPMMM9lN0HQLHrVp2yQ">
> Also, here's Chase Ackerman's profile: <img alt="image" width="1513" height="530" src="https://private-user-images.githubusercontent.com/122561188/517070681-ce90dbc7-bf78-41f9-8292-d5d800334ac4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM3Mzc0MjksIm5iZiI6MTc2MzczNzEyOSwicGF0aCI6Ii8xMjI1NjExODgvNTE3MDcwNjgxLWNlOTBkYmM3LWJmNzgtNDFmOS04MjkyLWQ1ZDgwMDMzNGFjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMVQxNDU4NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NGU4OTcyNDEzMDQ3NjUxMzlkM2Q1MDI2MDk5NWI5ZDVjZDRhZDAwNmUyMTYyZjhiMTc3ZDIzODk4MzY2ZWI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.AHGEXguHYTQzZCiQyGev28qeiyk7xNzA5bcXwpOxMGk">

That is a great point. We handle the rawClassLevel as an SQL NULL, rather than the string we had to catch that edge case. Now it is also working for syakira and other rawClassLevel = NULL people. It populates the number of terms as 8, as they have attended 8 terms to graduate.

Demonstration:
<img width="1001" height="341" alt="image" src="https://github.com/user-attachments/assets/8fb28051-b179-4675-906d-cbd03a7fd5f6" />

RueHaile committed:
Populating correctly for grguated students

ojmakinde commented:
@RueHaile, your image says "Bonner Orientation mine 0/8". There's only 1 Bonner Orientation event, if I remember correctly, and the "mine" is probably a typo. Additionally, I'm certain that Syakira did her Bonner Orientation and all the other required events, so there's a problem of a data mismatch happening somewhere in our data.

It's likely that the events were not tagged as requirement matches, and in that case, we may need to scan through all the events attended, and find the matches. An alternative could also be writing a script that does that traversal for every student and populates the database. @BrianRamsay

RueHaile commented:
> @RueHaile, your image says "Bonner Orientation mine 0/8". There's only 1 Bonner Orientation event, if I remember correctly, and the "mine" is probably a typo. Additionally, I'm certain that Syakira did her Bonner Orientation and all the other required events, so there's a problem of a data mismatch happening somewhere in our data.
> 
> It's likely that the events were not tagged as requirement matches, and in that case, we may need to scan through all the events attended and find the matches. An alternative could also be writing a script that does that traversal for every student and populates the database. @BrianRamsay

@ojmakinde, the picture that you see above with the "mine" is mine, as the name implies, I created it, and so that is why she didn't attend any of it. If you reset the data to prod-backup, you can see that there is a bonner orientation of frequency of one, and not eight. I added the picture to demonstrate that the field is getting populated. The Bonner orientation has not been attended by her and almost all others. I will look into the data mismatch, as it is a possibility. 
<img width="1312" height="458" alt="image" src="https://github.com/user-attachments/assets/90addb00-16e0-4c7e-86bb-6b3e1a99034e" />

BrianRamsay committed:
Merge branch 'development' into bonnerCheckmarkProfile

RueHaile commented:
@ojmakinde: It seems like a lack of data, rather than a bug/data mismatch/ as we went in while the database was set to development.
PROOF: 
<img width="581" height="182" alt="image" src="https://github.com/user-attachments/assets/2db9b598-7c1d-4dfa-9158-0e715ad622e7" />
and we got:- 
<img width="1118" height="880" alt="image" src="https://github.com/user-attachments/assets/075bae13-6c86-499e-aec3-6e819b480dbc" />

WHICH means that no data is saved for the event of booner orientation other than for event_id= 476 & 708 (Doesn't have any participants), where  event_id = 476 is found to have data in terms of rsvp but not as eventattendees despite the addition of "utsa: a volunteer, she would be not get populated mark as bonner oriantation happens once for every stident on their freshman year.

MImran2002 committed:
make sure if the duplicate removing function doesn't remove the once frequency certification if completed is exist later in the list

MImran2002 committed:
its the print statements that were breaking

JohnCox2211 commented:
It appears that the credit for new requirements is not being added to user profiles when a new requirement is created.

<img width="437" height="302" alt="image" src="https://github.com/user-attachments/assets/3687f96b-bc7a-4d86-990b-0982dd85691f" />

<img width="926" height="336" alt="image" src="https://github.com/user-attachments/assets/a992f3e8-47f6-4f19-ba5d-274be647e551" />

<img width="872" height="847" alt="image" src="https://github.com/user-attachments/assets/fb4eb1ad-527b-4012-9d2f-9286ec0b9010" />


The mark was not initially showing up for the test, so I edited the event to be an orientation, which then appeared. When I switched it back to the test requirement, the orientation mark stayed, and there was still no mark for the test requirement. 

When I tried this again, there was no longer a mark by the orientation when I changed the requirement to test, and there was still no check next to the test requirement. I then added an additional Bonner event for All Bonner Meeting, and added Zach; the check for the requirement did not appear next to All Bonner Meeting on their profile either, so neither the test requirement nor the All Bonner Meeting requirement appears. 

<img width="429" height="718" alt="image" src="https://github.com/user-attachments/assets/94eb537f-6987-47bb-93a0-90c7f0a8cda1" />

MImran2002 commented:
> It appears that the credit for new requirements is not being added to user profiles when a new requirement is created.
> 
> <img alt="image" width="437" height="302" src="https://private-user-images.githubusercontent.com/143194918/520989514-3687f96b-bc7a-4d86-990b-0982dd85691f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTg5NTE0LTM2ODdmOTZiLWJjN2EtNGQ4Ni05OTBiLTA5ODJkZDg1NjkxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MzIwYjFlNzEzMzllMzBhMjYzMjgyZGYxNjhjODRlNmQ0MjM3Y2JkMGY0NjRkNjNlMjZlYWIzZTA1NjY0OWM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.AoKUT4C8KnUGXH7hUPckPATyY6xiwjw5-3lG9L8Ufqw"> <img alt="image" width="926" height="336" src="https://private-user-images.githubusercontent.com/143194918/520989638-a992f3e8-47f6-4f19-ba5d-274be647e551.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTg5NjM4LWE5OTJmM2U4LTQ3ZjYtNGYxOS1iYTVkLTI3NGJlNjQ3ZTU1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTkxNGYyZTlkZThmMmZiMWVmNDcxZWFjYmRiMTBiNmFmOWFjNzYwNjRiZmMwZDlmYzU2YjUyMTdmYmE1NTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cxxwBS5jermLP4VF8LtYDiMxyf39irmCQ-MzYhJUHtA"> <img alt="image" width="872" height="847" src="https://private-user-images.githubusercontent.com/143194918/520990027-fb4eb1ad-527b-4012-9d2f-9286ec0b9010.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTkwMDI3LWZiNGViMWFkLTUyN2ItNDAxMi05ZDJmLTkyODZlYzBiOTAxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MDRiYTlhZmMxZTY1ZTgwYTEyOGQ4MGQyMzA1NzBjZTI5NjAzNTRiYWE4MjEyZTAzOTIzNDhjNjdiNGY3MjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.X-4tdtyH4cmCAJRVJXELQDSZMLGoVb_3MVaVJneAlc0">
> The mark was not initially showing up for the test, so I edited the event to be an orientation, which then appeared. When I switched it back to the test requirement, the orientation mark stayed, and there was still no mark for the test requirement.
> 
> When I tried this again, there was no longer a mark by the orientation when I changed the requirement to test, and there was still no check next to the test requirement. I then added an additional Bonner event for All Bonner Meeting, and added Zach; the check for the requirement did not appear next to All Bonner Meeting on their profile either, so neither the test requirement nor the All Bonner Meeting requirement appears.
> 
> <img alt="image" width="429" height="718" src="https://private-user-images.githubusercontent.com/143194918/520995673-94eb537f-6987-47bb-93a0-90c7f0a8cda1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTk1NjczLTk0ZWI1MzdmLTY5ODctNDdiYi05M2EwLTkwYzdmMGE4Y2RhMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZTkzMGEyMDI0ZTA5Y2U5ZjIwODY5MDhmODkyMzUyOTY5ZDVkZGQ0MTAyMjU5NmNiYjY4NjAzNmU1YjdjMDQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8Is9v7eYFfysYszadRudid1zjEpD2K_LPlKVJNStuX4">

Hi John, I have throughly looked through what you said definitely it is something to think about, I looked at and found the bug but the bug depends what the current term is in test the current term is flexible and we can assigned because of this line of code which it does is find all terms where the student have attended while the total term populate it base on the current term. Lets say the event is created at Spring 2025 while the current term is Spring 2021 because of the line: attendedTerms = set(term for term in attendedTerms  if term in totalTerms) it will only accept terms that are with the four year range below Spring 2021. That is why the bug exist. We can definitely talk about this at work and hopefully the bug recreation i did is like yours and this is the issue.  If this is the case then we should look into if it is an actual bug or more of a administrative edge case they wouldn't happen in real life. We should definitely talk to @ojmakinde !

JohnCox2211 commented:
> > It appears that the credit for new requirements is not being added to user profiles when a new requirement is created.
> > <img alt="image" width="437" height="302" src="https://private-user-images.githubusercontent.com/143194918/520989514-3687f96b-bc7a-4d86-990b-0982dd85691f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTg5NTE0LTM2ODdmOTZiLWJjN2EtNGQ4Ni05OTBiLTA5ODJkZDg1NjkxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MzIwYjFlNzEzMzllMzBhMjYzMjgyZGYxNjhjODRlNmQ0MjM3Y2JkMGY0NjRkNjNlMjZlYWIzZTA1NjY0OWM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.AoKUT4C8KnUGXH7hUPckPATyY6xiwjw5-3lG9L8Ufqw"> <img alt="image" width="926" height="336" src="https://private-user-images.githubusercontent.com/143194918/520989638-a992f3e8-47f6-4f19-ba5d-274be647e551.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTg5NjM4LWE5OTJmM2U4LTQ3ZjYtNGYxOS1iYTVkLTI3NGJlNjQ3ZTU1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTkxNGYyZTlkZThmMmZiMWVmNDcxZWFjYmRiMTBiNmFmOWFjNzYwNjRiZmMwZDlmYzU2YjUyMTdmYmE1NTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cxxwBS5jermLP4VF8LtYDiMxyf39irmCQ-MzYhJUHtA"> <img alt="image" width="872" height="847" src="https://private-user-images.githubusercontent.com/143194918/520990027-fb4eb1ad-527b-4012-9d2f-9286ec0b9010.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTkwMDI3LWZiNGViMWFkLTUyN2ItNDAxMi05ZDJmLTkyODZlYzBiOTAxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MDRiYTlhZmMxZTY1ZTgwYTEyOGQ4MGQyMzA1NzBjZTI5NjAzNTRiYWE4MjEyZTAzOTIzNDhjNjdiNGY3MjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.X-4tdtyH4cmCAJRVJXELQDSZMLGoVb_3MVaVJneAlc0">
> > The mark was not initially showing up for the test, so I edited the event to be an orientation, which then appeared. When I switched it back to the test requirement, the orientation mark stayed, and there was still no mark for the test requirement.
> > When I tried this again, there was no longer a mark by the orientation when I changed the requirement to test, and there was still no check next to the test requirement. I then added an additional Bonner event for All Bonner Meeting, and added Zach; the check for the requirement did not appear next to All Bonner Meeting on their profile either, so neither the test requirement nor the All Bonner Meeting requirement appears.
> > <img alt="image" width="429" height="718" src="https://private-user-images.githubusercontent.com/143194918/520995673-94eb537f-6987-47bb-93a0-90c7f0a8cda1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTk1NjczLTk0ZWI1MzdmLTY5ODctNDdiYi05M2EwLTkwYzdmMGE4Y2RhMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZTkzMGEyMDI0ZTA5Y2U5ZjIwODY5MDhmODkyMzUyOTY5ZDVkZGQ0MTAyMjU5NmNiYjY4NjAzNmU1YjdjMDQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8Is9v7eYFfysYszadRudid1zjEpD2K_LPlKVJNStuX4">
> 
> Hi John, I have throughly looked through what you said definitely it is something to think about, I looked at and found the bug but the bug depends what the current term is in test the current term is flexible and we can assigned because of this line of code which it does is find all terms where the student have attended while the total term populate it base on the current term. Lets say the event is created at Spring 2025 while the current term is Spring 2021 because of the line: attendedTerms = set(term for term in attendedTerms if term in totalTerms) it will only accept terms that are with the four year range below Spring 2021. That is why the bug exist. We can definitely talk about this at work and hopefully the bug recreation i did is like yours and this is the issue. If this is the case then we should look into if it is an actual bug or more of a administrative edge case they wouldn't happen in real life. We should definitely talk to @ojmakinde !

When looking into this further, I noticed the same issue with the mark for an event not appearing next to the requirement on the student's page, even when the event was set to have happened and was attended before the current term. The current term is summer 2021, and I created the event for November 2020. I followed the rest of the steps previously done and still ended up with 0/3 on the created requirement and on All Bonner Meeting.

RueHaile commented:
> > It appears that the credit for new requirements is not being added to user profiles when a new requirement is created.
> > <img alt="image" width="437" height="302" src="https://private-user-images.githubusercontent.com/143194918/520989514-3687f96b-bc7a-4d86-990b-0982dd85691f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTg5NTE0LTM2ODdmOTZiLWJjN2EtNGQ4Ni05OTBiLTA5ODJkZDg1NjkxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MzIwYjFlNzEzMzllMzBhMjYzMjgyZGYxNjhjODRlNmQ0MjM3Y2JkMGY0NjRkNjNlMjZlYWIzZTA1NjY0OWM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.AoKUT4C8KnUGXH7hUPckPATyY6xiwjw5-3lG9L8Ufqw"> <img alt="image" width="926" height="336" src="https://private-user-images.githubusercontent.com/143194918/520989638-a992f3e8-47f6-4f19-ba5d-274be647e551.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTg5NjM4LWE5OTJmM2U4LTQ3ZjYtNGYxOS1iYTVkLTI3NGJlNjQ3ZTU1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTkxNGYyZTlkZThmMmZiMWVmNDcxZWFjYmRiMTBiNmFmOWFjNzYwNjRiZmMwZDlmYzU2YjUyMTdmYmE1NTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cxxwBS5jermLP4VF8LtYDiMxyf39irmCQ-MzYhJUHtA"> <img alt="image" width="872" height="847" src="https://private-user-images.githubusercontent.com/143194918/520990027-fb4eb1ad-527b-4012-9d2f-9286ec0b9010.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTkwMDI3LWZiNGViMWFkLTUyN2ItNDAxMi05ZDJmLTkyODZlYzBiOTAxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MDRiYTlhZmMxZTY1ZTgwYTEyOGQ4MGQyMzA1NzBjZTI5NjAzNTRiYWE4MjEyZTAzOTIzNDhjNjdiNGY3MjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.X-4tdtyH4cmCAJRVJXELQDSZMLGoVb_3MVaVJneAlc0">
> > The mark was not initially showing up for the test, so I edited the event to be an orientation, which then appeared. When I switched it back to the test requirement, the orientation mark stayed, and there was still no mark for the test requirement.
> > When I tried this again, there was no longer a mark by the orientation when I changed the requirement to test, and there was still no check next to the test requirement. I then added an additional Bonner event for All Bonner Meeting, and added Zach; the check for the requirement did not appear next to All Bonner Meeting on their profile either, so neither the test requirement nor the All Bonner Meeting requirement appears.
> > <img alt="image" width="429" height="718" src="https://private-user-images.githubusercontent.com/143194918/520995673-94eb537f-6987-47bb-93a0-90c7f0a8cda1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2MzIzNzksIm5iZiI6MTc2NDYzMjA3OSwicGF0aCI6Ii8xNDMxOTQ5MTgvNTIwOTk1NjczLTk0ZWI1MzdmLTY5ODctNDdiYi05M2EwLTkwYzdmMGE4Y2RhMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIwMVQyMzM0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZTkzMGEyMDI0ZTA5Y2U5ZjIwODY5MDhmODkyMzUyOTY5ZDVkZGQ0MTAyMjU5NmNiYjY4NjAzNmU1YjdjMDQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8Is9v7eYFfysYszadRudid1zjEpD2K_LPlKVJNStuX4">
> 
> Hi John, I have throughly looked through what you said definitely it is something to think about, I looked at and found the bug but the bug depends what the current term is in test the current term is flexible and we can assigned because of this line of code which it does is find all terms where the student have attended while the total term populate it base on the current term. Lets say the event is created at Spring 2025 while the current term is Spring 2021 because of the line: attendedTerms = set(term for term in attendedTerms if term in totalTerms) it will only accept terms that are with the four year range below Spring 2021. That is why the bug exist. We can definitely talk about this at work and hopefully the bug recreation i did is like yours and this is the issue. If this is the case then we should look into if it is an actual bug or more of a administrative edge case they wouldn't happen in real life. We should definitely talk to @ojmakinde !

Note: There are edge cases that we need to account for when testing, such as the fact that certain students may not be present for specific semesters. As @MImran2002  said, we need to find a term that the student has attended while also making sure that, based on the current term that is set on the manage tab, it is a viable option as a term .

MImran2002 committed:
found a bug that could cause problem with none, null, graduating and non-degree- fixed it

MImran2002 committed:
fixed the bug and remove print statements

Karina-Agliullova reviewed (dismissed):
PR looks good! Tested it thoroughly and the code looks clean. Good job!

MImran2002 committed:
Merge branch 'development' of https://github.com/BCStudentSoftwareDevTeam/celts into bonnerCheckmarkProfile

MImran2002 committed:
adjusting what to show when the current term is summer
